templates/console/source/lambda/config-version/cf_config_version_manager/cf_config_version_manager.py

```diff
@@ -322,29 +278,65 @@
 
                 logger.info('target distributions been updated')
 
-    return {
-        'statusCode': 200,
-        'body': 'succeed apply snapshot to target distributions'
-    }
+
+def download_version_config(ddb_table, source_dist_id, src_version):
+    response = ddb_table.get_item(
+        Key={
+            "distributionId": source_dist_id,
+            "versionId": int(src_version)
+        })
+    data = response['Item']
+    s3_bucket = data['s3_bucket']
+    s3_key1 = data['s3_key']
+    s3_client = boto3.client('s3')
+    local_config_file_name_version = '/tmp/' + source_dist_id + "_" + str(src_version) + ".json"
+    s3_client.download_file(s3_bucket, s3_key1, local_config_file_name_version)
+    return local_config_file_name_version
+
+
+def get_snapshot_version(ddb_snapshot_table, snapshot_name, source_dist_id):
+    response = ddb_snapshot_table.get_item(
+        Key={
+            "distributionId": source_dist_id,
+            "snapShotName": snapshot_name
+        })
+    snapshot_resp = response['Item']
+    if not snapshot_resp:
+        raise Exception(
+            f"Failed to get the snapshot with distribution id:{source_dist_id}, snapshot_name:{snapshot_name}")
+    src_version = snapshot_resp['versionId']
+    logger.info(f"source version is {src_version}")
+    return src_version
+
+
+def validate_input_parameters(source_dist_id, src_snapshot, target_distribution_ids):
+    for dist_name in target_distribution_ids:
+        if dist_name == "" or dist_name.isspace():
+            raise Exception("target distribution list contains empty string")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.



Similar issue at line numbers 317 and 319.